hack/local-up-cluster.sh

```diff
@@ -217,6 +217,7 @@ LOG_LEVEL=${LOG_LEVEL:-3}
 # Use to increase verbosity on particular files, e.g. LOG_SPEC=token_controller*=5,other_controller*=4
 LOG_SPEC=${LOG_SPEC:-""}
 LOG_DIR=${LOG_DIR:-"/tmp"}
+TMP_DIR=${TMP_DIR:-$(mktemp -d 2>/dev/null || mktemp -d -t test-cluster.XXXXXX)}
```

Comment:
This approach still needs to be understood. if we don't find proper explanation of why it's used, I'd use `TMP_DIR=${TMP_DIR:-$(mktemp -d -t test-cluster.XXXXXXXXXX)}`

Reply:
@mhmxs let's use [this approach](https://github.com/kubernetes/kubernetes/pull/115106#discussion_r1072441528) suggested by @thockin

Reply:
I did @bart0sh, thanks for the review